vars/parseStageInfo.groovy

```diff
@@ -267,6 +267,26 @@ def call(Map config = [:]) {
       }
     }
 
+    String provider
+    // Highest priority is TestProvider parameter
+    if (startedByUser() && params.TestProvider && params.TestProvider != "") {
+      provider = params.TestProvider
+    } else {
+      // Next highest priority is a stage specific Test-provider-*
+      provider = cachedCommitPragma("Test-provider" + result['pragma_suffix'], null)
+      if (!repeat) {
+        // Followed by the more general Test-provider:
+        provider = cachedCommitPragma("Test-provider", null)
+      }
+    }
+    if (provider) {
+      if (result['ftest_arg']) {
+        result['ftest_arg'] += " --provider='" + provider + "'"
+      } else {
+        result['ftest_arg'] += "--provider='" + provider + "'"
+      }
```

Comment:
```suggestion
      if (result['ftest_arg']) {
        result['ftest_arg'] += ' '
      }
      result['ftest_arg'] += "--provider='" + provider + "'"
```
or
```suggestion
      result['ftest_arg'] += result['ftest_arg'] ? '' : ' ' + "--provider='" + provider + "'"
```
but maybe better still would be to just always add the trailing ' ' on every `result['ftest_arg'] += ` and then `.trim()` it at the end.  Or create a temporary list and then join it into a string at the end.

Also wonder how this squares with https://github.com/daos-stack/pipeline-lib/blob/2dca7b4d389dcb74de7ff34ee343b3e817ca46f3/vars/parseStageInfo.groovy#L306-L308.  Is it understood that any `ftest_arg` value passed by the caller will void adding `--provider` (and indeed, `--repeat`) by the pragmas or parameters here?

Reply:
That's fine.  I just wanted to point out the potential here.  If you are OK with it, it is fine with me also.